### ISSUE TITLE
Initialization of the database gives Errno 150

### ISSUE THREAD

**Issue by a20kimar** [HGustavs/LenaSYS#11310]
When initializing Lenasys with install.php it gets stuck and aborts when creating tables user and partresult because these tables declares foreign key on column level rather than table level in the file init_db.sql.

While trying to solve this I reached the conclusion that it might have to do with my version of MariaDB being 10.5.15 and most (if not all) other students use XAMPP which is 10.4.24. I have looked through the config files(for the database) and there seem to be nothing to affect how a foreign keys is declared. 

Removing these foreign keys solves the problem but still it's uncertain if this affects the site elsewhere.

**Comment by a20behta** [comment]
@a20kimar Branch created can start checking on the issue

**Comment by a20kimar** [comment]
I've found that there's two faulty foreign key declarations in the database. These are never initialized and skipped in older version of MariaDB(or some magic with XAMPP) while the newer versions recognize the issue and aborts with Errno 150. 

I took help of @a20caros in order to compare different databases.

Here's a example of how it looked before 
_Line: 422 /Shared/SQL/init_db.sql_
```
CREATE TABLE partresult (
 ...
	hp DECIMAL(3,1) **REFERENCES subparts (parthp)**,
	PRIMARY KEY (partname, cid, uid),
	FOREIGN KEY (partname,cid) REFERENCES subparts (partname,cid),
	FOREIGN KEY (uid) REFERENCES user (uid)
) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE=InnoDB;
```

which yield this result in the database regardless what version of MySQL you use.

<img width="787" alt="Screen Shot 2022-04-05 at 11 49 43" src="https://user-images.githubusercontent.com/82010032/161727793-cfeca0c9-cea8-44e6-b458-a153c4e47ad4.png">

I also find no documentation of foreign keys being declared column-level without using constraints as documentation states it should be. I found one instance of a declaration like this where the author stated that nothing is created by this kind of in-line foreign key creation with reference alone(as shown above surrounded by asterix).

Therefore my conclusion is that these two can safely be removed and nothing will happen except the fact that future installations of LenaSYS will not stumble where mine did.

For testing:
Run this after initializing both a older version of the database and this merge and compare the two. There should be no difference between the two. Change 'partresult' to 'user' to check both tables that are affected by this change.
SELECT * FROM information_schema.TABLE_CONSTRAINTS  WHERE information_schema.TABLE_CONSTRAINTS.CONSTRAINT_TYPE = 'FOREIGN KEY'  AND information_schema.TABLE_CONSTRAINTS.TABLE_NAME = 'partresult';
